src/scanner-cli.ts

```diff
@@ -51,13 +52,19 @@ export function normalizePlatformName(): 'windows' | 'linux' | 'macosx' {
 /**
  * Where to download the SonarScanner CLI
  */
-function getScannerCliUrl(properties: ScannerProperties, version: string): URL {
+function getScannerCliUrl(properties: ScannerProperties, versionStr: string): URL {
   // Get location to download scanner-cli from
 
   // Not in default to avoid populating it for non scanner-cli users
   const scannerCliMirror = properties[ScannerProperty.SonarScannerCliMirror] ?? SCANNER_CLI_MIRROR;
+  const version = semver.coerce(versionStr);
+  if (!version) {
+    throw new Error(`Version "${versionStr}" does not have a correct format."`);
+  }
+  const arch = version.compare('6.1.0') >= 0 ? '-x64' : '';
```

Comment:
Are we sure it's fine to use the `x64` scanner-cli on ARM? Do we need a ticket to try and download the `aarch64` version when needed?

+ see [this file](https://github.com/SonarSource/sonar-scanner-npm/blob/master/src/platform.ts#L24)

Reply:
That's what we were doing before 6.1.0, using the `-x64` flavor (the only available). We could create a separate ticket to use the ARM when needed on Linux and Mac, but IMO this is a separate task.

Reply:
Note that according to the NodeJS doc, we will have the same issue that with Java. The value returned by `process.arch` is not the real Architecture of the CPU, but the flavor of the NodeJS runtime. So for people on Rosetta emulation, using NodeJS x64, it will return `x64`, and so we will use bootstrap the scanner/JRE x64.